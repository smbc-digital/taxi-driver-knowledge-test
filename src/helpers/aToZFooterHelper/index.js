const showAToZFooter = (show) =>{
    let footer  = document.getElementById('atoz')
    if(footer !== null) 
    {
        if(show){
            footer.classList.remove('hidden')
        }
        else{
            footer.classList.add('hidden')
        }
    }
}

export default showAToZFooter