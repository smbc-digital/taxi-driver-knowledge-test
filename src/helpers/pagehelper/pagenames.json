[
	{
		"number": 1,
		"route": "/book-taxi-driver-knowledge-test/test-type"
	},
	{
		"number": 2,
		"route": "/book-taxi-driver-knowledge-test/first-time"
	},
	{
		"number": 3,
		"route": "/book-taxi-driver-knowledge-test/about-yourself"
	},
	{
		"number": 4,
		"route": "/book-taxi-driver-knowledge-test/choose-appointment"
	},
	{
		"number": 5,
		"route": "/book-taxi-driver-knowledge-test/booking-summary"
	},
	{
		"number": 6,
		"route": "/book-taxi-driver-knowledge-test/success"
	},
	{
		"number": 7,
		"route": "/book-taxi-driver-knowledge-test/something-went-wrong"
	},
	{
		"number": 8,
		"route": "/book-taxi-driver-knowledge-test/failed-booking"
	},
	{
		"number": 9,
		"route": "/book-taxi-driver-knowledge-test/payment-declined"
	}
]