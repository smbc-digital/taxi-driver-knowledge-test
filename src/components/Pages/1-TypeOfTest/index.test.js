import { React, mount} from '../../../helpers/SetupTest'
import { TypeOfTest } from './index'
import { getPageRoute } from '../../../helpers/pagehelper'
import renderer from 'react-test-renderer'

describe('TypeOfTest', () => {
    it('should call push on submit', () => {
        //Arrange
        const data = {
            testType: {
                value: 'Private hire',
                isValid: true
            }
        }

        const history = { push: jest.fn() }
		const wrapper = mount(<TypeOfTest context={data} history={history} />)

        //Act
        wrapper.find('form').simulate('submit')
        //Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(2))
    })

    describe('snapshot', () => {
		it('should render correctly',() => {
			// Arrange
			const data = {
                testType: {
                    value: 'Private Hire',
                    isValid: true
                }
			}

			// Act
			const tree = renderer
			.create(<TypeOfTest context={data} />)
			.toJSON()

			// Assert
			expect(tree).toMatchSnapshot()
		})
	})
})
