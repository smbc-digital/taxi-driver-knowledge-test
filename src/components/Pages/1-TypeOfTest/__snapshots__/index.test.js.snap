// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeOfTest snapshot should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1 />
  <h2>
    Which taxi driver knowledge test do you want to take?
  </h2>
  <fieldset>
    <div
      className="input-radio-container"
    >
      <input
        checked={true}
        disabled={undefined}
        id="private"
        name="testType"
        onChange={[Function]}
        type="radio"
        value="Private hire"
      />
      <label
        htmlFor="private"
      >
        Private hire
      </label>
    </div>
    <div
      className="input-radio-container"
    >
      <input
        checked={undefined}
        disabled={undefined}
        id="hackney"
        name="testType"
        onChange={[Function]}
        type="radio"
        value="Hackney carriage"
      />
      <label
        htmlFor="hackney"
      >
        Hackney carriage
      </label>
    </div>
    <div
      className="input-radio-container"
    >
      <input
        checked={undefined}
        disabled={undefined}
        id="executive"
        name="testType"
        onChange={[Function]}
        type="radio"
        value="Executive hire"
      />
      <label
        htmlFor="executive"
      >
        Executive hire
      </label>
    </div>
    <div
      className="input-radio-container"
    >
      <input
        checked={undefined}
        disabled={undefined}
        id="assisted"
        name="testType"
        onChange={[Function]}
        type="radio"
        value="Assisted transport"
      />
      <label
        htmlFor="assisted"
      >
        Assisted transport
      </label>
    </div>
  </fieldset>
  <button
    className="button-primary"
    disabled={false}
    type="submit"
  >
    Next step
  </button>
</form>
`;
